src/Functions/splitByChar.cpp

```diff
@@ -77,12 +78,12 @@ class SplitByCharImpl
 
     bool get(Pos & token_begin, Pos & token_end)
     {
-        if (!pos)
+        if (!pos) [[unlikely]]
             return false;
 
         token_begin = pos;
 
-        if (max_splits)
+        if (max_splits) [[unlikely]]
```

Comment:
It feels like it should lead to a degradation in the case where `max_splits` will be used. It cannot be said to be unlikely, because it is a constant after initialization. It should be predicted well by the processor IMO.

Reply:
fixed.